tests/rptest/tests/raft_availability_test.py

```diff
@@ -421,9 +442,26 @@ def test_follower_isolation(self):
                 FailureSpec(FailureSpec.FAILURE_ISOLATE,
                             self.redpanda.get_node(replicas[1])))
 
-            # expect messages to be produced and consumed without a timeout
-            for i in range(0, 128):
-                self._ping_pong()
+            nodes = admin.get_redpanda_nodes()
+            nodes = [node for node in nodes if node.id != replicas[1]]
+            admin.await_stable_leader(
+                "id_allocator",
+                namespace="kafka_internal",
+                replication=3,
+                timeout_s=ELECTION_TIMEOUT * 2,
+                nodes=nodes,
+                check=lambda node: node.id != replicas[1])
+
+            count = 0
+            while count < 128:
+                # a request may fail when a init producer request lands on
```

Comment:
Is this type of failure specific to when idempotency is enabled?

I think the original test expectation is that if the client tries to talk to the isolated node, then it should time out and retry quickly enough to succeed within the overall 5s timeout.

(I think @mmaslankaprv wrote this one, so might remember better the original intent)

Reply:
The problem was that an idempotent related rpc used to return an non retry-able error, I've changed it but a client still chooses a node to connect to randomly at best so there is a chance that it may time out before it finds the right node. I've increased the timeout, let's see if it helps.

In the worst case (a client sticks to an isolated node) we'll retry on the app (test) level.